### PR TITLE
ODC-7667: Migrate SecureRouteFields to PF5

### PR DESCRIPTION
**Pull request by logonoff**
**Fixes**: 

https://issues.redhat.com/browse/ODC-7667

**Solution Description**: 

Migrate file to PF5

**Screen shots / Gifs for design review**: 

Before:
image

After:
image

Demo:

https://github.com/user-attachments/assets/6789a3da-93ff-4b67-ae16-76bec020b616

**Unit test coverage report**: 

unchanged

**Test setup:**

n/a

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari

### CODE REVIEW

File: frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx [hunk 1]
```diff
@@ -145,38 +189,30 @@ const SecureRouteFields: React.FC = () => {
...
-          placeholderText={t('devconsole~Select termination type')}
-          isDisabled={!preferredRoutingOptionsLoaded}
-          aria-label={t('devconsole~Select termination type')}
-          maxHeight={300}
+          selected={tlsTermination}
+          onOpenChange={onTLSTerminationToggle}
+          toggle={tlsTerminationToggle}
+          shouldFocusToggleOnSelect
         >
           {tlsTerminationSelectOptions}
```
Comment (rhamilto) [inline frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx:201]: Add `<SelectList>`

File: frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx [hunk 2]
```diff
@@ -145,38 +189,30 @@ const SecureRouteFields: React.FC = () => {
...
           onSelect={onInsecureTrafficSelect}
-          placeholderText={t('devconsole~Select insecure traffic type')}
-          isDisabled={!preferredRoutingOptionsLoaded}
-          aria-label={t('devconsole~Select insecure traffic type')}
-          maxHeight={300}
+          onOpenChange={onInsecureTrafficToggle}
+          toggle={insecureTrafficToggle}
+          shouldFocusToggleOnSelect
         >
           {insecureTrafficSelectOptions}
```
Comment (rhamilto) [inline frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx:214]: Add `<SelectList>`

**Comment by logonoff** [comment]
Tech debt that is not customer facing, so manually adding labels
/label px-approved
/label docs-approved

**Comment by yanpzhan** [comment]
Checked on cluster against the pr, the SecureRouteFields works as before. 
@logonoff @rhamilto maybe it's not related to this pr, but prompt here, I'm doubted about the layout and text color on the "Applications" page: why some texts are in gray color(compared to texts under other tab, eg Language tab)? And why indents exist only before "Routes can be secured using several TLS termination types for serving certificates."(compared to other substring).

**Review by vikram-raj — APPROVED** [review]
/lgtm

**Pull request merged**